File: backend/app/rag/chat_modes.py
"""Chat mode prompt hints (Sprint 3)."""
from __future__ import annotations

CHAT_MODES = ("default", "mentor", "beginner", "debate", "reviewer")

CHAT_MODE_SYSTEM_HINTS: dict[str, str] = {
    "default": "保持准确、基于引用回答。回答优先结构化：结论、关键依据、相关文档。",
    "mentor": (
        "你是一位严谨的研究导师。在回答中："
        "（1）指出 USER_QUESTION 中潜在的假设、变量未控制、文献局限或可证伪点；"
        "（2）必要时反问以澄清研究目标；"
        "（3）给出 2-3 个下一步实验或读什么的建议。"
        "不得编造证据；必要时说明 CONTEXT 不足。"
    ),
    "beginner": (
        "你面对的是研究新手。回答需："
        "（1）使用类比和直觉解释，不要堆术语；"
        "（2）首次出现的术语在括号内简短解释；"
        "（3）结尾给出 1 条 \"如果想深入可以读哪篇\" 的推荐。"
    ),
    "debate": (
        "你以辩论结构回答，明确分两栏：「支持观点」「反对观点」。"
        "每条论点必须有 CONTEXT 中的引用。"
        "最后给出一段中性总结，不要替用户下结论。"
    ),
    "reviewer": (
        "你以匿名审稿人的视角回答。逐项评估："
        "（1）方法贡献是否充分；（2）实验是否能支撑结论；"
        "（3）局限与潜在威胁；（4）改进建议。"
        "严格但不刻薄，禁止用 \"X 错了\" 这类绝对判定。"
    ),
}


def mode_hint(mode: str | None) -> str:
    return CHAT_MODE_SYSTEM_HINTS.get(mode or "default", CHAT_MODE_SYSTEM_HINTS["default"])


__all__ = ["CHAT_MODES", "CHAT_MODE_SYSTEM_HINTS", "mode_hint"]

File: frontend/src/utils/chatModes.ts
import type { ChatMode } from "../types/api";

/** Shared between the per-topic ChatPanel and the CrossTopicChatPage so we have
 *  one source of truth for chat-mode labels + 1-line descriptions (used in the
 *  selector tooltip).
 *
 *  Must stay in sync with `backend/app/rag/chat_modes.py`. */
export interface ChatModeMeta {
  value: ChatMode;
  label: string;
  desc: string;
}

export const CHAT_MODES: ChatModeMeta[] = [
  { value: "default", label: "默认", desc: "结构化回答:结论、关键依据、相关文档。" },
  { value: "mentor", label: "导师", desc: "挑出问题的假设/局限,反问澄清,给出下一步建议。" },
  { value: "beginner", label: "入门", desc: "用类比解释,术语带括号注解,推荐 1 篇深入阅读。" },
  { value: "debate", label: "辩论", desc: "支持/反对两栏并各自带引用,最后中性总结。" },
  { value: "reviewer", label: "审稿人", desc: "逐项评估:方法贡献、实验是否支撑、局限、改进。" },
  { value: "what_if", label: "假设", desc: "反事实推理:基线 → 如果 X → 推动 Y,标出最薄弱假设。" },
];

export const CHAT_MODE_DESC: Record<ChatMode, string> = CHAT_MODES.reduce(
  (acc, m) => {
    acc[m.value] = m.desc;
    return acc;
  },
  {} as Record<ChatMode, string>,
);
